### ISSUE TITLE
scalars in metadata/actions.graphql

### ISSUE THREAD

**Issue by tomasherout** [hasura/graphql-engine#8450]
### Version Information

Server Version: 2.6.0
CLI Version (for CLI related issue): 2.6.0

### Environment

OSS

### What is the expected behaviour?

File `metadata/actions.graphql` should contains scalars definitions.

Without them file can't be used for code generation with tools like [GraphQL Code Generator](https://www.graphql-code-generator.com)

### Keywords

actions.graphql scalars

### What is the current behaviour?

Scalars used in actions are not defined in file `metadata/actions.graphql`

### How to reproduce the issue?

Run `hasura export` command in hasura-cli.

### Screenshots or Screencast

<img width="362" alt="image" src="https://user-images.githubusercontent.com/43199492/165938234-a6098fe5-f517-4f0d-b203-7e4cc950a38d.png">

### Any possible solutions?

Include scalars definitions for types used in actions like:

```graphql

scalar uuid

type Mutation {
  addPerons(
    personId: uuid
  ): String!
}

```

### Can you identify the location in the source code where the problem exists?

No

### If the bug is confirmed, would you be willing to submit a PR?

I am not familiar with Haskell

**Comment by tomasherout** [comment]
Yes, but **schema file** `metadata/actions.graphql` is invalid without defined scalars. I don't think define them manually in Hasura Console.

In case file `metadata/actions.graphql` includes all non-built in GraphQL scalars (other than Int, Float, String, Boolean and ID) than it can be used for other purposes (like code generation). Currently the file is invalid for this purpose:

```
 % graphql-codegen --config graphql/hasura-actions/codegen.yml 
(node:3212) ExperimentalWarning: stream/web is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
  ✔ Parse configuration
  ❯ Generate outputs
    ❯ Generate graphql/hasura-actions/generated.ts
      ✖ Load GraphQL schemas
        → Failed to load schema
        Load GraphQL documents
        Generate


 Found 1 error

  ✖ graphql/hasura-actions/generated.ts
    Failed to load schema from hasura/metadata/actions.graphql:

        Unknown type: "uuid".
        Error: Unknown type: "uuid".
```

**Comment by kdawgwilk** [comment]
This also applies to the new experimental support for Apollo Federation where the gateway can ask hasura for its schema sdl e.g. `query { _service { sdl } }` but the schema returned from this endpoint is invalid because it does not contain the scalar types used in the schema
```
/usr/app/node_modules/@apollo/gateway/src/supergraphManagers/IntrospectAndCompose/index.ts:118
      throw Error(
            ^
Error: A valid schema couldn't be composed. The following composition errors were found:
	[hasura] Unknown type uuid
```

**Comment by tirumaraiselvan** [comment]
@kdawgwilk Thanks for raising this. Do you mind raising a new ticket for this issue with Apollo Federation integration specifically? The original ask on Actions, although quite similar in essence, is in a very different module.

**Comment by kdawgwilk** [comment]
Yes I opened an issue here #9115

**Comment by lukaso** [comment]
BTW: This issue is now biting me when upgrading my schema from 2.8.0 to 2.33.2.

The postgress uuid scalar is no longer auto applied, meaning my schema is in an inconsistent state.

The fix is to do the following:

To `actions.graphql` add anywhere (I added to the top):

```
scalar uuid
```

To `actions.yaml` replace in the `custom_types` key:

```
custom_types:

  ...

  scalars: []
```

Change to:

```
  scalars:
    - name: uuid
```

I imagine the same would be true for the postgres `geography` scalar.
